### PR TITLE
Support pagination of organizations on user settings pages

### LINKED ISSUE #16076: Pagination in user settings/orgniazaion
As title.

### PR DESCRIPTION
**Pull request by lunny**
Fix #16076

- [x] Support pagniation of organizations on user settings pages
- [x] Support pagniation of organizations on API
- [x] Remove function `GetOrgsByUserID` and use `FindOrgs`

### CODE REVIEW

**Comment by lunny** [comment]
depends on #16099

**Review by zeripath — APPROVED** [review]

**Comment by zeripath** [comment]
CI fail is related

**Comment by 6543** [comment]
there are two ways to indicate if you are part of an org: `org_user` and `team_user + team` tables !

fixtures are not complete consistent & we should drop org_user table in the log rung and just use the team to check against!
if this will consume to much resources we at least make sure it stay consistent :/

cc @lunny this is a refactor who will need a migration, so it's for a separate dedicated pull :)

**Comment by 6543** [comment]
ok looks like we have to go the "make fixtures consistent & provide check mechanism for doctor ..."

**Comment by 6543** [comment]
did some changes but it still fails :/ -> https://github.com/lunny/gitea/pull/19#issuecomment-861788821

the generated sql looks valide and this particular part isn fistures is correct too ... so no idear what's wrong

**Review by wxiaoguang — APPROVED** [review]

**Review by 6543 — APPROVED** [review]

File: models/org.go [hunk 1]
```diff
@@ -489,6 +489,49 @@ func GetUserOrgsList(user *User) ([]*MinimalOrg, error) {
...
+type FindOrgOptions struct {
+	db.ListOptions
+	UserID         int64
+	IncludePrivate bool
+}
+
+func (opts FindOrgOptions) toConds() builder.Cond {
+	var cond = builder.NewCond()
+	if opts.UserID > 0 {
+		// TODO: queryUserOrgIDs do not exist anymore
```
Comment (6543) [inline models/org.go]: @lunny this todo has to be resolved :)

Comment (lunny) [reply models/org.go]: done.

**Comment by 6543** [comment]
@lunny 

```
--- FAIL: TestFindOrgs (0.02s)
    org_test.go:333: 
        	Error Trace:	org_test.go:333
        	Error:      	"[]" should have 1 item(s), but has 0
        	Test:       	TestFindOrgs
    org_test.go:349: 
        	Error Trace:	org_test.go:349
        	Error:      	Not equal: 
        	            	expected: int(1)
        	            	actual  : int64(0)
        	Test:       	TestFindOrgs
```

**Comment by zeripath** [comment]
CI fail is related

**Review by wxiaoguang — APPROVED** [review]

**Pull request merged**